src/group_theory/amenable/mean.lean

```diff
@@ -0,0 +1,399 @@
+          : by exact m.lin_map.map_smul' M _
+    ... = M • 1
+          : by congr'; exact m.normality
+    ... = M
+          : by simp,
+  },
+
+
+  have : m f + m diff = M
+  := by
+  calc  m f + m diff
+      = m (f + diff)
+        : by exact (m.lin_map.map_add' f diff).symm
+  ... = m (f + bounded_continuous_function.const G M - f)
+        : by simp[diff]
+  ... = m (bounded_continuous_function.const G M )
+        : by simp
+  ... = M
+        : by simp [mean_const],
+
+  by linarith only [this, mdiffpos],
+end
+
+
+
+/--Essentially: W.r.t. the sup-norm, m has norm ≤ 1-/
+lemma mean_bounded_abs
+  (m : mean G)
+  {f: bounded_continuous_function G ℝ}
+  {M : ℝ}
+  (fbound : ∀ (x:G), |f x| ≤ M)
+  : |m f| ≤ M
+:= begin
+  have bound_le : m f ≤ M,
+  {
+    have fbound' :  ∀ (x:G), f x ≤ M
+      := (λ x, (abs_le.mp (fbound x)).2),
+    exact mean_bounded G m fbound',
+  },
+
+  have bound_ge : m f ≥ -M,
+  {
+    have negfbound' :  ∀ (x:G), (-f) x ≤ M,
+    {
+      assume x:G,
+      simp,
+      by linarith[(abs_le.mp (fbound x)).1],
+    },
+    have : m (-f) ≤ M
+            := mean_bounded G m negfbound',
+    have : m (-f) = - m f
+            := mean_of_neg G m,
+    by linarith,
+  },
+
+  exact abs_le.mpr (and.intro bound_ge bound_le),
+end
+
+
+@[simp]
+lemma mean_add
+  {m : mean G}
+  {f g: bounded_continuous_function G ℝ}
+  : m (f+g) = m f + m g
+:= m.lin_map.map_add' f g
+
+
+
+@[simp]
+lemma mean_smul
+  {m : mean G}
+  {f: bounded_continuous_function G ℝ}
+  {r :ℝ}
+  : m (r•f) =r •  (m f)
+:= m.lin_map.map_smul' r f
+
+
+/--Means are monotone functions-/
+lemma mean_monotone
+  {m : mean G}
+  {f g: bounded_continuous_function G ℝ}
+  (f_le_g : f ≤ g)
+  : m f ≤ m g
+:= begin
+  have diff_pos: ∀ (x:G), (g-f) x ≥ 0,
+  {
+    assume x:G,
+    have : (g-f) x = g x - f x
+      := by refl,
+    rw this,
+    simp,
+    exact f_le_g x,
+  },
+  calc  m f
+      = m f + 0
+        : by ring
+  ... ≤  m f + m (g-f)
+        : by {simp, exact m.positivity diff_pos,}
+  ... = m (f+(g-f))
+        : by rw mean_add
+  ... = m g
+        : by congr';ring,
+end
+
+
+
+
+
+end el_facts
+
+
+
+section pushforward_mean
+
+/-!
+### Pushforwards of Means
+
+We will often use the following construction: If `m` is a mean on `H`
+and `π : G → H` is any map, we can obtain a mean on `G`
+by composing with `π`.
+
+-/
+
+
+variables {H : Type* } [uniform_space H] [group H] [topological_group H]
+(π: G → H)
+(π_cont: continuous π)
+
+variable {G}
+
+/-- Precomposition of a bounded_continuous_map with a continuous map-/
+def bcont_precomp
+  {X Y: Type*}
+  [topological_space X]
+  [topological_space Y]
+  (h : C(X,Y))
+  (f : bounded_continuous_function Y ℝ)
+  : bounded_continuous_function X ℝ
+:= bounded_continuous_function.mk
+      (continuous_map.comp f.to_continuous_map h)
+      (by {
+        rcases f.map_bounded' with ⟨C, hf⟩,
+        use C,
+        assume x y : X,
+        specialize hf (h x) (h y),
+        exact hf,
+      })
```

Comment:
This exists already, your `bcont_precomp h f` is `f.comp_continuous h`

Reply:
Thanks, I replaced this now